drift/core/psestimation.py

```diff
@@ -625,16 +627,55 @@ def q_estimator(self, mi, vec1, vec2=None, noise=False):
         x1 = np.dot(evecs.T.conj(), x0)
 
         # Project back into sky basis
-        x2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(mi, x1, conj=True)
+        x2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(
+            mi, x1, temponly=True, conj=True
+        )
+        # Slice array for temponly
+        x2 = x2[:, 0]
 
         if vec2 is not None:
             y0 = (vec2.T / (evals + 1.0)).T
             y1 = np.dot(evecs.T.conj(), x0)
-            y2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(mi, x1, conj=True)
+            y2 = self.kltrans.beamtransfer.project_vector_svd_to_sky(
+                mi, x1, temponly=True, conj=True
+            )
+            # Slice array for temponly
+            y2 = y2[:, 0]
+
         else:
             y0 = x0
             y2 = x2
 
+        return x2, y2
+
+    def q_estimator(self, mi, vec1, vec2=None, noise=False):
```

Comment:
Given that you have an entirely new implementation of `q_estimator` in your new class, why does the old one need to get changed?

Reply:
I wanted to factor out the part of the code that projects from KL-basis to sky basis so that it can be reused as a method in the future.

Reply:
and the new PS-code would get just like a big spaghetti code. Tried to factor out as much as I could.

Reply:
Yep, makes sense to me